enigma/src/main/java/cuchaz/enigma/analysis/StructureTreeNode.java

```diff
@@ -0,0 +1,151 @@
+package cuchaz.enigma.analysis;
+
+import cuchaz.enigma.EnigmaProject;
+import cuchaz.enigma.api.service.NameProposalService;
+import cuchaz.enigma.api.service.ObfuscationTestService;
+import cuchaz.enigma.translation.TranslateResult;
+import cuchaz.enigma.translation.mapping.EntryRemapper;
+import cuchaz.enigma.translation.representation.TypeDescriptor;
+import cuchaz.enigma.translation.representation.entry.*;
+
+import javax.swing.tree.DefaultMutableTreeNode;
+import java.util.List;
+
+public class StructureTreeNode extends DefaultMutableTreeNode {
+    private final List<NameProposalService> nameProposalServices;
+    private final EntryRemapper mapper;
+    private final ClassEntry parentEntry;
+    private final ParentedEntry entry;
+
+    public StructureTreeNode(EnigmaProject project, ClassEntry parentEntry, ParentedEntry entry) {
+        this.nameProposalServices = project.getEnigma().getServices().get(NameProposalService.TYPE);
+        this.mapper = project.getMapper();
+        this.parentEntry = parentEntry;
+        this.entry = entry;
+    }
+
+    /**
+     * Returns the parented entry corresponding to this tree node.
+     */
+    public ParentedEntry getEntry() {
+        return this.entry;
+    }
+
+    public void load(EnigmaProject project, boolean hideDeobfuscated) {
+        List<ParentedEntry> children = project.getJarIndex().getChildrenByClass().get(this.parentEntry);
+
+        for (ParentedEntry child : children) {
+            StructureTreeNode childNode = new StructureTreeNode(project, this.parentEntry, child);
+
+            if (child instanceof ClassEntry) {
+                childNode = new StructureTreeNode(project, (ClassEntry) child, child);
+                childNode.load(project, hideDeobfuscated);
+            }
+
+            // don't add deobfuscated members if hideDeobfuscated is true, unless it's an inner class
+            if (hideDeobfuscated && this.isDeobfuscated(project, child) && !(child instanceof ClassEntry)) {
+                continue;
+            }
+
+            // don't add constructor methods if hideDeobfuscated is true
+            if (hideDeobfuscated && (child instanceof MethodEntry) && ((MethodEntry) child).isConstructor()) {
+                continue;
+            }
+
+            this.add(childNode);
+        }
+    }
+
+    private boolean isDeobfuscated(EnigmaProject project, ParentedEntry child) {
+        List<ObfuscationTestService> obfuscationTestServices = project.getEnigma().getServices().get(ObfuscationTestService.TYPE);
```

Comment:
It isn't very important- but is this similar logic replicated anywhere else? It might be nice to be able to deduplicate it since this logic doesn't seem unique to here.

Reply:
There was the same logic in StatsGenerator, moved it to EnigmaProject